File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
       HashMap<Integer,Integer>numsMap = new HashMap<>();
       int i = 0 ;
       while(i<nums.length){
        int needed = target - nums[i];
        if(numsMap.containsKey(needed)){
            return new int []{
                numsMap.get(needed),i 
            };
        }
            numsMap.put(nums[i],i);
            i++ ; 
       }
        return new int []{} ;
    }  
}

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
   let i=0
   const map = new Map()
   while(i<nums.length){
    let needed = target - nums[i]
    if(map.has(needed))return [map.get(needed),i]
    map.set(nums[i],i)
    i++
   }
return []
}